### ISSUE TITLE
Docs builder container image is rebuilt everytime

### ISSUE THREAD

**Issue by gdemonet** [scality/metalk8s#2192]
**Component**: build

**What happened**:

Whenever we trigger a CI run with some changes anywhere in the repo, the `docs-builder` container image will be rebuilt.

**What was expected**:

It should only be rebuilt if strict dependencies required to **build** the image were changed.

**Resolution proposal** (optional):

The Docker build `context` used for this image is set to the root of the repository, so we can include files such as `VERSION`.

Some possible solutions:

- if we had Docker 19 with BuildKit, this would allow using various `<name>.dockerignore` files for different images (see moby/moby/issues/12886, and [this comment](https://github.com/moby/moby/issues/12886#issuecomment-518843764) in particular) - however, I don't think our CI uses this version yet (cc @tcarmet ?)
- we could instead duplicate the files we need in `docs/`, and have a linting step that checks consistency between duplicated files
- build the docs as part of the buildchain, and do some magic tricks to make it work (e.g. add symlinks for the out-of-context files and use a [tarball context](https://docs.docker.com/engine/reference/commandline/build/#tarball-contexts) for the build)

I would prefer the first option, but it may take a long time to implement, so this remains open for discussion.

**Comment by slaperche-scality** [comment]
> build the docs as part of the buildchain

It's already the case, no?

I mean, we're building the PDF for the ISO, and the HTML for the UI container so we are already building the doc in the buildchain.

That may be not true for all the branches, but the "build doc"-related code may not be too big to backport.

**Comment by gdemonet** [comment]
Oh, so we could get rid of this `docs` stage in our `pre-merge` CI test suite?

**Comment by slaperche-scality** [comment]
Yeah I think we could (but not in all branches, except if we backport code).

I didn't do it when I introduced the "doc-in-buildchain" code because at first we were only building the PDF. But now we're building both, so it's probably doable to use what's generated under `_build/docs`.
